src/runner.py

```diff
@@ -99,34 +131,39 @@ def run(self):
         try:
             while True:
                 checkout_node = self._db.receive_node(sub_id)
-
-                self._logger.log_message(logging.INFO, "Tarball: {}".format(
-                    checkout_node['artifacts']['tarball']
-                ))
-
-                self._logger.log_message(logging.INFO, "Creating test node")
-                node = self._create_node(checkout_node)
-
-                tmp = tempfile.TemporaryDirectory(dir=self._output)
-                output_file = self._generate_job(node, tmp.name)
-
-                self._logger.log_message(logging.INFO, "Running test")
-                process = self._runtime.submit(output_file, get_process=True)
+                process, tmp = self._schedule_test(checkout_node)
                 self._job_tmp_dirs[process] = tmp
-
                 self._cleanup_paths()
         except KeyboardInterrupt as e:
             self._logger.log_message(logging.INFO, "Stopping.")
         finally:
             self._db.unsubscribe(sub_id)
             self._cleanup_paths()
 
+    def run(self):
+        if self._node_id:
+            checkout_node = self._db.get_node(node_id)
+            self._run_single_job(self._checkout_node)
```

Comment:
```suggestion
            self._run_single_job(checkout_node)
```

Or else we can directly write,
```
self._run_single_job(self._db.get_node(self._node_id))
```
I think the first approach is better for code readability.

Reply:
I think 2 lines is better in this case. Thanks I'll fix it.